### DISCUSSION TITLE
http-api-server and sharelink

### DISCUSSION THREAD

**Discussion by chbister** [avantrec/soco-cli#37] [Q&A] [answered]
Hi @pwt ,

nice piece of software, thanks!

When using the http-api-server, I am not able to add a sharelink (e.g. the one you provided in the docs: https://music.apple.com/dk/album/black-velvet/217502930?i=217503142)

http://api.example.com:8000/Dachgeschoss/sharelink/https%3A%2F%2Fmusic.apple.com%2Fdk%2Falbum%2Fblack-velvet%2F217502930%3Fi%3D217503142
Always returns { "detail": "Not Found" }

It works with the command line tool without issue:
sonos Dachgeschoss sharelink https://music.apple.com/dk/album/black-velvet/217502930?i=217503142

What am I doing wrong? Do I need to encode the link or somehow escape it?

(I guess git is a better place for questions than this: https://en.community.sonos.com/controllers-software-228995/sonos-command-line-controller-available-python-soco-cli-6844948/index3.html?postid=16610913#post16610913)

**Comment by pwt** [comment, marked as answer]
The URL isn't being handled correctly, so the HTTP server is trying to find the path internally. This is the log at the server end:
```
INFO:     127.0.0.1:63186 - "GET /move/sharelink/https%3A//music.apple.com/dk/album/black-velvet/217502930%3Fi%3D217503142 HTTP/1.1" 404 Not Found
```

I'll take a look at this when I get a moment. Issue raised: #38.

**Reply by chbister** [reply]
This is exactly what I see. Thanks a lot!

**Reply by pwt** [reply]
Should be fixed in v0.4.51, which I've just released. Let me know if you see any issues.

**Reply by chbister** [reply]
Is fixed in v0.4.51. That was quick!!!

SoCo-CLI: Command = 'sonos Dachgeschoss sharelink https://music.apple.com/dk/album/black-velvet/217502930', exit code = 0
INFO:     192.168.199.173:58718 - "GET /Dachgeschoss/sharelink/https%3A//music.apple.com/dk/album/black-velvet/217502930?i=217503142 HTTP/1.1" 200 OK
